nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/excel/ExcelRecordReader.java

```diff
@@ -0,0 +1,221 @@
+            LAZY_TIME_FORMAT = () -> DataTypeUtils.getDateFormat(timeFormat);
+        }
+
+        if (isEmpty(args.getTimestampFormat())) {
+            this.timestampFormat = null;
+            LAZY_TIMESTAMP_FORMAT = null;
+        } else {
+            this.timestampFormat = args.getTimestampFormat();
+            LAZY_TIMESTAMP_FORMAT = () -> DataTypeUtils.getDateFormat(timestampFormat);
+        }
+
+        try {
+            this.rowIterator = new RowIterator(args.getInputStream(), desiredSheets, args.getFirstRow(), logger);
+        } catch (RuntimeException e) {
+            String msg = "Error occurred while processing record file";
+            logger.error(msg, e);
+            throw new MalformedRecordException(msg, e);
+        }
+    }
+
+    @Override
+    public Record nextRecord(boolean coerceTypes, boolean dropUnknownFields) throws IOException, MalformedRecordException {
+        try {
+            if (rowIterator.hasNext()) {
+                Row currentRow = rowIterator.next();
+                Map<String, Object> currentRowValues = getCurrentRowValues(currentRow, coerceTypes, dropUnknownFields);
+                return new MapRecord(schema, currentRowValues);
+            }
+        } catch (Exception e) {
+            throw new MalformedRecordException("Error while getting next record", e);
+        }
+        return null;
+    }
+
+    private Map<String, Object> getCurrentRowValues(Row currentRow, boolean coerceTypes, boolean dropUnknownFields) {
+        final List<RecordField> recordFields = schema.getFields();
+        final Map<String, Object> currentRowValues = new LinkedHashMap<>();
+
+        if (ExcelUtils.hasCells(currentRow)) {
+            IntStream.range(0, currentRow.getLastCellNum())
+                    .forEach(index -> {
+                        Cell cell = currentRow.getCell(index);
+                        Object cellValue;
+                        if (index >= recordFields.size()) {
+                            if (!dropUnknownFields) {
+                                cellValue = getCellValue(cell);
+                                currentRowValues.put("unknown_field_index_" + index, cellValue);
+                            }
+                        } else {
+                            final RecordField recordField = recordFields.get(index);
+                            String fieldName = recordField.getFieldName();
+                            DataType dataType = recordField.getDataType();
+                            cellValue = getCellValue(cell);
+                            final Object value = coerceTypes ? convert(cellValue, dataType, fieldName)
+                                    : convertSimpleIfPossible(cellValue, dataType, fieldName);
+                            currentRowValues.put(fieldName, value);
+                        }
+                    });
+        }
+
+        return currentRowValues;
+    }
+
+    public static Object getCellValue(Cell cell) {
```

Comment:
Is there a reason this method is public? If it is public for testing, it probably should be moved to a utility class.